kanidmd_web_ui/src/components/change_unix_password.rs

```diff
@@ -0,0 +1,237 @@
+use compact_jwt::{Jws, JwsUnverified};
+use kanidm_proto::v1::{SingleStringRequest, UserAuthToken};
+use std::str::FromStr;
+use wasm_bindgen::{JsCast, JsValue, UnwrapThrowExt};
+use wasm_bindgen_futures::JsFuture;
+use web_sys::{FormData, HtmlFormElement};
+
+use web_sys::{Request, RequestInit, RequestMode, Response};
+use yew::prelude::*;
+
+use crate::error::*;
+use crate::utils;
+
+pub struct ChangeUnixPassword {
+    state: State,
+}
+
+#[derive(Debug, Default)]
+struct FormValues {
+    password_input: String,
+    password_repeat_input: String,
+}
+
+impl From<FormData> for FormValues {
+    fn from(data: FormData) -> Self {
+        Self {
+            password_input: data.get("password_input").as_string().unwrap(),
+            password_repeat_input: data.get("password_repeat_input").as_string().unwrap(),
+        }
+    }
+}
+
+pub enum Msg {
+    Submit(FormData),
+    Error { emsg: String, kopid: Option<String> },
+    Success,
+}
+
+impl From<FetchError> for Msg {
+    fn from(fe: FetchError) -> Self {
+        Msg::Error {
+            emsg: fe.as_string(),
+            kopid: None,
+        }
+    }
+}
+
+pub enum State {
+    Init,
+    Error { emsg: String, kopid: Option<String> },
+}
+
+#[derive(PartialEq, Eq, Properties)]
+pub struct ChangeUnixPasswordProps {
+    pub token: String,
+}
+
+impl Component for ChangeUnixPassword {
+    type Message = Msg;
+    type Properties = ChangeUnixPasswordProps;
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        Self { state: State::Init }
+    }
+    fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Msg::Submit(data) => {
+                let fd: FormValues = data.into();
+                if fd.password_input != fd.password_repeat_input {
+                    return self.update(
+                        ctx,
+                        Msg::Error {
+                            emsg: "Password fields did not match".to_string(),
+                            kopid: None,
+                        },
+                    );
+                }
+                let tk = ctx.props().token.clone();
+                ctx.link().send_future(async {
+                    match Self::update_unix_password(tk, fd.password_input).await {
```

Comment:
Worth checking the password update code in the other credential update path since I think this can return a password quality errer from zxcvbn, and maybe there is a way to share some code between the two?

Reply:
While implementing the blocking of the submit button, I figured out just how similar the two modals are. I'll probably work on a shared component after this is merged.